### PR TITLE
Restore BehaviorSubject behavior for `HydratedSubject`

### PR DESCRIPTION
**Pull request by solid-yuriiprykhodko**

### CODE REVIEW

File: lib/src/hydrated.dart [hunk 1]
```diff
@@ -137,40 +115,37 @@ class HydratedSubject<T> extends Subject<T> implements ValueStream<T> {
 
   @override
   void onAdd(T event) {
-    _wrapper = SubjectValueWrapper(value: event);
+    _subject.value = event;
```
Comment (solid-vovabeloded) [inline lib/src/hydrated.dart:118]: Would it be a bit more readable?
```suggestion
    _subject.add(event);
```

File: lib/src/hydrated.dart [hunk 2]
```diff
@@ -107,28 +100,13 @@ class HydratedSubject<T> extends Subject<T> implements ValueStream<T> {
...
-      onCancel: onCancel,
-      sync: sync,
-    );
-
-    final wrapper =
-        seedValue != null ? SubjectValueWrapper(value: seedValue) : null;
-
+    final subject = seedValue != null
+        ? BehaviorSubject<T>.seeded(seedValue,
+            onListen: onListen, onCancel: onCancel, sync: sync)
```
Comment (solid-vovabeloded) [inline lib/src/hydrated.dart:105]: Would it be great to add trailing commas for these constructors?

File: test/hydrated_test.dart [hunk 3]
```diff
@@ -75,6 +77,39 @@ void main() {
     /// clean up
     subject.close();
   });
+
+  test('Listener BehaviorSubject behavior', () async {
```
Comment (solid-vovabeloded) [inline test/hydrated_test.dart:81]: What do you think about adding more context into the test description. For example: 
`HydratedSubject emits the latest value on listener added`.

File: test/hydrated_test.dart [hunk 4]
```diff
@@ -75,6 +77,39 @@ void main() {
...
+    await completer.future;
+
+    expect(
+        subject.stream,
+        emits(
+          isA<SerializedClass>().having((c) => c.value, 'value', isTrue),
+        ));
+
+
+    final expectation = expectLater(
```
Comment (solid-vovabeloded) [inline test/hydrated_test.dart:100]: I think there is no sense in double-checking that the stream emits the latest value.

File: test/hydrated_test.dart [hunk 5]
```diff
@@ -75,6 +77,39 @@ void main() {
...
+    final completer = Completer();
+
+    final subject = HydratedSubject<SerializedClass>(
+      "SerializedClass",
+      hydrate: (s) => SerializedClass.fromJSON(s),
+      persist: (c) => c.toJSON(),
+      onHydrate: () => completer.complete(),
+    );
+
+    await completer.future;
```
Comment (solid-vovabeloded) [inline test/hydrated_test.dart:91]: It seems like we can use `subject.first` instead of this completer.

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Great job, @solid-yuriiprykhodko! I've added a couple of suggestions here - please take a look. Overall, the code looks great - thanks.

File: .github/workflows/flutter.yaml [hunk 6]
```diff
@@ -13,4 +13,4 @@ jobs:
       - run: flutter pub get
       - run: flutter analyze
       - name: Run tests
-        run: flutter test
\ No newline at end of file
+        run: flutter test -r expanded
```
Comment (solid-yuriiprykhodko) [inline .github/workflows/flutter.yaml:16]: This makes CI log formatting a bit nicer -- we see all the test cases that were executed.

File: lib/src/hydrated.dart [hunk 7]
```diff
@@ -1,12 +1,10 @@
 import 'dart:async';
 
 import 'package:flutter/foundation.dart';
-import 'package:hydrated/src/utils/typing_utils.dart';
+import 'package:hydrated/src/utils/type_utils.dart';
```
Comment (solid-yuriiprykhodko) [inline lib/src/hydrated.dart:4]: Missed this in the previous PR -- the class was renamed, but the file wasn't.

**Review by solid-vovabeloded — DISMISSED** [review]
LGTM, thanks @solid-yuriiprykhodko!

**Pull request merged**